tests/update_workflow_test.go

```diff
@@ -2795,91 +2765,56 @@ func (s *UpdateWorkflowSuite) TestUpdateWorkflow_SpeculativeWorkflowTask_Schedul
 
 	tv = s.startWorkflow(tv)
 
-	wtHandlerCalls := 0
-	wtHandler := func(task *workflowservice.PollWorkflowTaskQueueResponse) ([]*commandpb.Command, error) {
-		wtHandlerCalls++
-		switch wtHandlerCalls {
-		case 1:
-			// Completes first WT with empty command list.
-			return nil, nil
-		case 2:
-			// Speculative WT timed out on sticky task queue. Server sent full history with sticky timeout event.
-			s.EqualHistory(`
-  1 WorkflowExecutionStarted
-  2 WorkflowTaskScheduled
-  3 WorkflowTaskStarted
-  4 WorkflowTaskCompleted
-  5 WorkflowTaskScheduled // Speculative WT.
-  6 WorkflowTaskTimedOut
-  7 WorkflowTaskScheduled {"Attempt":1} // Normal WT.
-  8 WorkflowTaskStarted`, task.History)
-			return nil, nil
-		default:
-			s.Failf("wtHandler called too many times", "wtHandler shouldn't be called %d times", wtHandlerCalls)
-			return nil, nil
-		}
-	}
-
-	msgHandlerCalls := 0
-	msgHandler := func(task *workflowservice.PollWorkflowTaskQueueResponse) ([]*protocolpb.Message, error) {
-		msgHandlerCalls++
-		switch msgHandlerCalls {
-		case 1:
-			return nil, nil
-		case 2:
-			updRequestMsg := task.Messages[0]
-			// Reject update, but WT still will be in the history due to timeout on sticky queue.
-			return s.UpdateRejectMessages(tv, updRequestMsg, "1"), nil
-		default:
-			s.Failf("msgHandler called too many times", "msgHandler shouldn't be called %d times", msgHandlerCalls)
-			return nil, nil
-		}
-	}
-
-	poller := &testcore.TaskPoller{
-		Client:                       s.FrontendClient(),
-		Namespace:                    s.Namespace(),
-		TaskQueue:                    tv.TaskQueue(),
-		StickyTaskQueue:              tv.StickyTaskQueue(),
-		StickyScheduleToStartTimeout: 1 * time.Second, // Important!
-		Identity:                     tv.WorkerIdentity(),
-		WorkflowTaskHandler:          wtHandler,
-		MessageHandler:               msgHandler,
-		Logger:                       s.Logger,
-		T:                            s.T(),
-	}
-
 	// Drain first WT and respond with sticky enabled response to enable sticky task queue.
-	_, err := poller.PollAndProcessWorkflowTask(testcore.WithRespondSticky, testcore.WithoutRetries)
+	stickyScheduleToStartTimeout := 1 * time.Second
+	_, err := s.PollAndProcessWorkflowTask(tv,
+		func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondWorkflowTaskCompletedRequest, error) {
+			return &workflowservice.RespondWorkflowTaskCompletedRequest{
+				StickyAttributes: tv.StickyExecutionAttributes(stickyScheduleToStartTimeout),
+			}, nil
+		})
```

Comment:
If you leave `taskpoller.DrainTaskFunc` as you have it now (ignoring my previous comment), and add `SetStickyOption` then it still can be a one liner.

Reply:
I had an option like that before while I developed it; but it felt it was hiding too much. 🤔

Reply:
Extracting this to `taskpoller.SetSticky(stickyScheduleToStartTimeout)` still looks attractive to me, but I don't insist.

Reply:
I'd prefer to leave it explicit for now. Once we have more use cases, we can re-consider.